client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -226,12 +233,15 @@ const Checkout = React.createClass( {
 			}
 
 			return purchasePaths.managePurchase( renewalItem.extra.purchaseDomain, renewalItem.extra.purchaseId );
-		} else if ( cartItems.hasFreeTrial( this.props.cart ) ) {
+		} else if ( cartItems.hasFreeTrial( cart ) ) {
 			this.props.clearSitePlans( selectedSiteId );
 
 			return selectedSiteSlug
 				? `/plans/${ selectedSiteSlug }/thank-you`
 				: '/checkout/thank-you/plans';
+		} else if ( selectedSite.options.is_domain_only && cartItems.hasDomainRegistration( cart ) && ! cartItems.hasPlan( cart ) ) {
```

Comment:
Should we use the selector (`client/state/selectors/is-domain-only-site.js`) that @gziolo created in #10656?